.idea/checkstyle-idea.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CheckStyle-IDEA">
```

Comment:
@olamy yes, that is the CheckStyle-IDEA plugin that I'm using.
The file is using relative paths so the risk of being out-of-sync should be minimal, but I'm not going to make a fuss about it and I can remove it and instead add a README entry about it.

Reply:
I just found [this](https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems), so this is the recommended way to share IDEA configuration between developers.

This is project-specific so it should hurt to have it there and I think it will make contributors life easier, because the project will be auto-configured .

Reply:
@olamy do you want me to revent all changes in .idea directory and not commit it there?
I can add a README entry for code formatting setup instead or we can allow this in the repo as explained in the link above.